tests/functional/tests/test_ceph.py

```diff
@@ -1,24 +1,48 @@
 """Test deployment and functionality of juju-verify."""
 import logging
+import re
+import time
 
+import zaza
 from juju import loop
 from tests.base import BaseTestCase
 
 from juju_verify import juju_verify
+from juju_verify.utils.action import cache
 from juju_verify.verifiers import get_verifier
+from juju_verify.verifiers.result import Result
 
 logger = logging.getLogger()
 
 
-class CephTests(BaseTestCase):
-    """Functional testing for Ceph verifier."""
+class CephOsdTests(BaseTestCase):
+    """Functional testing for ceph-osd verifier."""
 
-    # To add:
-    # test not OK for two ceph-osd unit stops
-    # test we get a warning when a machine also hosts another unit
+    test_pool = "test"
+
+    # TODO: add test to check_availability_zone after new approach
+
+    def _add_test_pool(self, percent_data: int = 10):
+        """Add a test pool."""
+        _ = zaza.model.run_action("ceph-mon/0", "create-pool",
+                                  action_params={"name": self.test_pool,
+                                                 "percent-data": percent_data})
+        # wait a bit so it settles
+        logger.debug("wait 60s for the test pool to settle...")
+        time.sleep(60)
```

Comment:
As discussed offline, this should be managed using `tenacity` to retry with exponential back off. Hopefully, checks that Ceph status is ok after creating a new test pool will last less than 60s each time (it may depend on the `percent_data` value and the resources of the test env).